### ISSUE TITLE
TYPO3 Install Tool not completely working

### ISSUE THREAD

**Issue by dhoffmann1979** [drud/ddev#985]
**Describe the bug**
Step 4 of the install tool (while first install) is shown. When Admincredentials are given to the form and the form is submitted, the user ist inserted in database, but the install tool jumps back to Step 4.

**To Reproduce**
see above

**Expected behavior**
The install tool should go to step 5

**Version and configuration information (please complete the following information):**
 - Host computer OS and Version: macOS High Sierra
 - Docker version information (use `docker version`) and insert it here.
Client:
 Version:      18.03.1-ce
 API version:  1.37
 Go version:   go1.9.5
 Git commit:   9ee9f40
 Built:        Thu Apr 26 07:13:02 2018
 OS/Arch:      darwin/amd64
 Experimental: false
 Orchestrator: swarm

Server:
 Engine:
  Version:      18.03.1-ce
  API version:  1.37 (minimum version 1.12)
  Go version:   go1.9.5
  Git commit:   9ee9f40
  Built:        Thu Apr 26 07:22:38 2018
  OS/Arch:      linux/amd64
  Experimental: true
 - ddev version information (use `ddev version`)
domain	ddev.local                      
cli   	v0.20.0                         
web   	drud/nginx-php-fpm-local:v0.20.0
db    	drud/mariadb-local:v0.20.0      
dba   	drud/phpmyadmin:v0.20.0         
router	drud/ddev-router:v0.20.0        
commit	v0.20.0             
 - config.yaml contents for the misbehaving project
APIVersion: v0.20.0
name: bul
type: typo3
docroot: html
php_version: "7.1"
router_http_port: "80"
router_https_port: "443"
xdebug_enabled: false
additional_hostnames: []
provider: default

 - Do you have any custom configuration (nginx, php, mysql) in the .ddev folder? If so, have you tried without them?
no extra files

**Comment by rfay** [comment]
Please say exactly what version of TYPO3 you're trying to install, thanks. Have you tried it in a different environment?

**Comment by rfay** [comment]
I should note that with TYPO3 8.7, and an older mac (2013 MacBook Air) I saw something that sounds like this. After chasing it quite a lot it appeared to me to be a timing bug in the TYPO3 8 install. So if you're seeing this on an older mac, and it's TYPO3 8, that's likely the issue. If it's what I saw, it doesn't happen on TYPO3 9, which has a completely new installer. It also doesn't happen on newer macs.

**Comment by dhoffmann1979** [comment]
it is TYPO3 8.7, it happens with and without composer installation. The mac is MacBook Air (13-inch, 2017)

**Comment by rfay** [comment]
If you haven't tried a reboot of your mac, please do. I've never been able to recreate the problem I saw on anything newer than 2013, and wow did I try. 

What browser are you using for the install?

**Comment by rfay** [comment]
I just ran through an install with a clean composer build of TYPO3 v8 (composer create-project typo3/cms-base-distribution typo3-v8-composer ^8) and was not able to recreate the problem you saw. I was using a Macbook Pro 2017. I confess to never having seen this on a Macbook Pro, and the only time I saw it (repeatedly) was on a 2013 air. So I suppose it could be something about the MBA rather than the year.  I did spend quite a bit of time chasing and debugging when it happened, and I couldn't come up with any answer other than a TYPO3 v8 timing-related bug in the install. And of course, the core devs can't be interested in that because v9 doesn't have the problem, and it's pretty obscure.

If you have the option, I recommend trying v9. 

My only theory is that the macbook Air  brings out the TYPO3 v8 problem due to  the slower disk access time (due to docker mount).

**Comment by rfay** [comment]
@dhoffmann1979 did you get past this? Would love to hear your experience.

**Comment by wowawebdesign** [comment]
I have the same problem when trying to install TYPO3 8.7 and 9.3.2. Maybe it's related to Docker, because a colleague of mine has the same problem in his Docker environment without using ddev.

**Comment by rfay** [comment]
If it's the same problem I saw, it would happen only in 8.7 and not in 9.3.2, seems to be a timing assumption in the TYPO3 install tool. 

@wowawebdesign can you give details about your physical host, operating system, and docker version please?

Thanks for the feedback @wowawebdesign

**Comment by wowawebdesign** [comment]
MacBook Pro 2017, i7, 16GB RAM, macOS High Sierra 10.13.5, Docker CE Version 18.03.1-ce-mac65 (24312), Docker Compose 1.21.1.

**Comment by wowawebdesign** [comment]
Problem still exists with Docker CE 18.06.0-ce-mac70 (26399) and Compose 1.22.0

**Comment by wowawebdesign** [comment]
Here's a short video: http://www.screencast.com/t/LvrERQAHm

After Step 4 nothing happens.

**Comment by rfay** [comment]
Perfect! Could you please disable your password program and do the same again? (Or try on another host?)

**Comment by wowawebdesign** [comment]
I have the same problem on an Ubuntu 18.04 without any password manager app.

**Comment by rfay** [comment]
With that password manager there, it looks for all the world like it's the problem. I understand that you were interacting with it so have more of an idea what was going on.

How about just using an incognito window here and do the exact same thing.

**Comment by wowawebdesign** [comment]
Ok, maybe I have the time for a better video this evening ;) But be sure, 1password is not the problem ;)

**Comment by wowawebdesign** [comment]
Ok, here's a new video.

OS: Ubuntu 18.04
DDEV 1.0.0
Docker 18.06.0-ce
docker-compose 1.22.0, build f46880fe

http://www.screencast.com/t/I1hRN1HRg

**Comment by rfay** [comment]
I can demonstrate this, but it looks to me like TYPO3 9.3.0 works fine. Are you sure this is working on 9.3.1 (which is what I assume you're getting?) We definitely need to sort out whether this is a TYPO3 or a ddev problem (or the combination of the two)

**Comment by rfay** [comment]
It works fine for me in 8.7.16 (latest version I had laying around)

**Comment by wowawebdesign** [comment]
Try 

composer create-project typo3/cms-base-distribution yourdirectory 8.7

for the latest 8.7

**Comment by wowawebdesign** [comment]
Ups, sorry! I tested it again right now with 8.7.17 and it worked!

That's strange, a few days ago the installation of 8.7.17 fails too! So it seems, the problem is only in the 9.x branch.

**Comment by rfay** [comment]
Having a hard time testing this - `composer create-project typo3/cms-base-distribution junktypo3 =9.2.1` starts installing 9.2.1: `Installing typo3/cms-base-distribution (v9.2.1)
  - Installing typo3/cms-base-distribution (v9.2.1): Loading from cache`

But then it *actually* installs 9.3.2 - thoughts? 

composer: https://gist.github.com/rfay/13bb893b91da22c5105928cf272f3d24

**Comment by rfay** [comment]
BTW, I invited you to the TYPO3 #ddev channel on TYPO3 slack.

**Comment by codemonkey1988** [comment]
I will try it this evening too. If I can reproduce this issue, I will do some debugging to provide more information about the root cause

**Comment by rfay** [comment]
Note that typo3/cms-base-distribution has a major error in that all versions install *latest*, they specify `"typo3/minimal": "^9.2",`, which of course picks up recent versions, not the 9.2 we would expect. So this is not easily debuggable using composer create-project.

**Comment by rfay** [comment]
Opened https://github.com/TYPO3/TYPO3.CMS.BaseDistribution/issues/8 for the problem with typo3/cms-base-distribution

**Comment by codemonkey1988** [comment]
It's a TYPO3 problem. See https://forge.typo3.org/issues/85552
Its because of the Doctrine 2.8 update. But it's fixed in current master.

**Comment by rfay** [comment]
Interesting, I saw the duplicate https://forge.typo3.org/issues/85588 go by, but hadn't heard about this one. 

So this was introduced between TYPO3 v9.3.0 (Jun 11) and v9.3.1 (Jul 12) (probably). v9.3.1 and v9.3.2 were on the same day, probably something went wrong. This issue was filed on July 10 though.  But the OP @dhoffmann1979 was actually having a problem with a Macbook Air that I have seen before in the 8.x install.

Not sure why a git checkout of 9.3.2 with a composer install would *not* show the problem.

But all version testing with composer create-project is suspect because of https://github.com/TYPO3/TYPO3.CMS.BaseDistribution/issues/8

I'm going to go ahead and close this as a duplicate of https://forge.typo3.org/issues/85552, but happy to continue the conversation here, and happy to reopen it.

**Comment by wowawebdesign** [comment]
Can confirm this: with the current TYPO3-Master everything works. Thanks for your patience and help!
